app/assets/src/components/views/compare/SamplesHeatmapVis.jsx

```diff
@@ -57,18 +70,24 @@ class SamplesHeatmapVis extends React.Component {
   }
 
   extractSampleLabels() {
-    return this.props.sampleIds.map(id => this.props.sampleDetails[id].name);
+    return this.props.sampleIds.map(id => {
+      return {
+        label: this.props.sampleDetails[id].name,
+        metadata: this.props.sampleDetails[id].metadata
+      };
+    });
   }
 
   extractTaxonLabels() {
-    return this.props.taxonIds.map(id => this.props.taxonDetails[id].name);
+    return this.props.taxonIds.map(id => {
+      return {
+        label: this.props.taxonDetails[id].name,
+        metadata: this.props.taxonDetails[id].metadata
```

Comment:
Only `sampleIds` have metadata, right? Do you need this for `taxonIds`?

Reply:
We will probably use taxonIds metadata too.... but not yet. This is when I was trying to make the code generic for that. But will remove.